eclair-core/src/test/scala/fr/acinq/eclair/integration/basic/ZeroConfAliasIntegrationSpec.scala

```diff
@@ -112,111 +110,123 @@ class ZeroConfAliasIntegrationSpec extends FixtureSpec with IntegrationPatience
     )
+
+    // Carol must use Bob's scid alias in her routing hints.
+    val carolHint = getRouterData(carol).privateChannels.values.head.toIncomingExtraHop.get
+    val bobAlias = getRouterData(bob).privateChannels.values.head.shortIds.localAlias
+    assert(carolHint.shortChannelId == bobAlias)
   }
 
-  test("a->b->c (b-c zero-conf unconfirmed private)", Tag(ZeroConfBobCarol)) { f =>
+  test("a->b->c (b->c zero-conf unconfirmed private)", Tag(ZeroConfBobCarol)) { f =>
+    import f._
+
     internalTest(f,
       deepConfirm = false,
       bcPublic = false,
       bcZeroConf = true,
       bcScidAlias = false,
-      paymentWorksWithoutHint = false, // alice can't find a route to carol because b-c isn't announced
+      paymentWorksWithoutHint = false, // alice can't find a route to carol because bob->carol isn't announced
       paymentWorksWithHint_opt = Some(true), // with a routing hint the payment works
-      paymentWorksWithRealScidHint_opt = None // there is no real scid for b-c yet
+      paymentWorksWithRealScidHint_opt = None // there is no real scid for bob->carol yet
     )
+
+    // Carol uses Bob's scid alias until the channel confirms.
+    val carolHint = getRouterData(carol).privateChannels.values.head.toIncomingExtraHop.get
+    val bobAlias = getRouterData(bob).privateChannels.values.collectFirst { case pc if pc.remoteNodeId == carol.nodeParams.nodeId => pc }.get.shortIds.localAlias
+    assert(carolHint.shortChannelId == bobAlias)
   }
 
-  test("a->b->c (b-c zero-conf deeply confirmed private)", Tag(ZeroConfBobCarol)) { f =>
+  test("a->b->c (b->c zero-conf deeply confirmed private)", Tag(ZeroConfBobCarol)) { f =>
     internalTest(f,
       deepConfirm = true,
       bcPublic = false,
       bcZeroConf = true,
       bcScidAlias = false,
-      paymentWorksWithoutHint = false, // alice can't find a route to carol because b-c isn't announced
+      paymentWorksWithoutHint = false, // alice can't find a route to carol because bob->carol isn't announced
       paymentWorksWithHint_opt = Some(true), // with a routing hint the payment works
-      paymentWorksWithRealScidHint_opt = None // skipped, see below
-      // TODO  This actually doesn't work, because the ChannelRelayer relies on the the LocalChannelUpdate event to maintain
-      // TODO  its scid resolution map, and the channel doesn't emit a new one when a real scid is assigned, because we use the
-      // TODO  remote alias for the channel_update, not the real scid. So the channel_update remains the same. We used to
-      // TODO  have the ChannelRelayer also listen to ShortChannelIdAssigned event, but it's doesn't seem worth it here.
+      // TODO: we should be able to send payments with the real scid in the routing hint, but this currently doesn't work,
+      //  because the ChannelRelayer relies on the the LocalChannelUpdate event to maintain its scid resolution map, and
```

Comment:
I'm not 100% sure we'll need to support that, it depends on the behavior of other implementations when they generate their routing hints in that case...we could argue that private channels that use 0-conf will most likely also support `scid_alias`, so this will never happen IRL?

Reply:
No I meant, TIL that you can have a multi-lines TODO by just indenting the subsequent lines.

I agree we don't need to implement relaying with real scids on a channel that we know has an alias